compiler/src/dotty/tools/dotc/transform/ReifyQuotes.scala

```diff
@@ -162,7 +184,6 @@ class ReifyQuotes extends MacroTransformWithImplicits {
     def tryHeal(tp: Type, pos: Position)(implicit ctx: Context): Option[String] = tp match {
       case tp: TypeRef =>
         if (level == 0) {
-          assert(ctx.owner.is(Macro))
```

Comment:
Why was the assert dropped?

Reply:
In one case where the macro contained an anonymous function inside and the `tryHeal` saw that function as `ctx.owner`. Now the assert is
```scala
assert(ctx.owner.ownersIterator.exists(_.is(Macro)))
```